quilt/test/test_push.py

```diff
@@ -37,24 +26,37 @@ def test_push(self):
         pkg_hash = pkg_obj.get_hash()
         assert pkg_hash
         contents = pkg_obj.get_contents()
-        urls = upload_urls(contents)
-        for url in urls.values():
-            self._mock_s3(url)
 
-        self._mock_put_package('foo/bar', pkg_hash, contents)
+        all_hashes = set(find_object_hashes(contents))
+        for blob_hash in all_hashes:
+            head_url = "https://example.com/head/{owner}/{hash}".format(
+                owner='foo', hash=blob_hash)
+            put_url = "https://example.com/put/{owner}/{hash}".format(
+                owner='foo', hash=blob_hash)
+            self._mock_get_blob('foo', blob_hash, head_url, put_url)
+            self._mock_s3(head_url, put_url)
+
+        self._mock_put_package('foo/bar', pkg_hash)
         self._mock_put_tag('foo/bar', 'latest')
 
         command.push('foo/bar')
 
-    def _mock_put_package(self, package, pkg_hash, contents):
-        pkg_url = '%s/api/package/%s/%s' % (command.QUILT_PKG_URL, package, pkg_hash)
-        self.requests_mock.add(responses.PUT, pkg_url, json.dumps(dict(
-            upload_urls=upload_urls(contents)
+    def _mock_get_blob(self, user, blob_hash, s3_head, s3_put):
+        blob_url = '%s/api/blob/%s/%s' % (command.QUILT_PKG_URL, user, blob_hash)
+        self.requests_mock.add(responses.GET, blob_url, json.dumps(dict(
+            head=s3_head,
+            get='',
+            put=s3_put,
         )))
 
+    def _mock_put_package(self, package, pkg_hash):
+        pkg_url = '%s/api/package/%s/%s' % (command.QUILT_PKG_URL, package, pkg_hash)
+        self.requests_mock.add(responses.PUT, pkg_url, json.dumps(dict()))
```

Comment:
It's the mock. The `dict()` is the server's response - which is not used by the CLI anymore.